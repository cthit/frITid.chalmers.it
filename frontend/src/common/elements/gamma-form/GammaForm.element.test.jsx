import React from "react";
import { shallow } from "enzyme";
import GammaForm from "./";

describe("<GammaForm/>", () => {
  test("Shallow render of <GammaForm/>", () => {
    const wrapper = shallow(
      <GammaForm
        onSubmit={(values, actions) => {
          console.log(values);
          console.log(actions);
        }}
        initialValues={{
          user: "asdf",
          password: "fdas",
          userAgreement: true,
          mySwitch: false,
          icecream: "chocolate",
          number: "5000"
        }}
        render={() => (
          <div>
            <GammaFormField
              name="user"
              component={GammaTextField}
              componentProps={{ lowerLabel: "User" }}
            />

            <GammaFormField
              name="password"
              component={GammaTextField}
              componentProps={{ upperLabel: "this is password yes" }}
            />

            <GammaFormField
              name="userAgreement"
              component={GammaCheckbox}
              componentProps={{
                primary: true,
                label: "Jag accepterar anvÃ¤ndaravtalen"
              }}
            />

            <GammaFormField
              name="mySwitch"
              component={GammaSwitch}
              componentProps={{
                label: "This is a switch",
                primary: true
              }}
            />

            <GammaFormField
              name="icecream"
              component={GammaSelect}
              componentProps={{
                upperLabel: "Favorite Icecream flavor",
                lowerLabel: "Choose the best flavor",
                valueToTextMap: {
                  chocolate: "Chocolate",
                  vanilla: "Vanilla",
                  strawberry: "Strawberry"
                },
                allowToChooseNone: true
              }}
            />

            <GammaFormField
              name="number"
              component={GammaRadioButtonGroup}
              componentProps={{
                upperLabel: "Upperlabel",
                lowerLabel: "Lowerlabel",
                radioButtonMap: {
                  "5000": {
                    primary: true,
                    label: "Primary"
                  },
                  "4000": {
                    secondary: true,
                    label: "Secondary"
                  },
                  "3000": {
                    label: "Default"
                  },
                  "2000": {
                    disabled: true,
                    label: "Disabled"
                  }
                }
              }}
            />
            <GammaFABButton primary component={Add} submit />
          </div>
        )}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
