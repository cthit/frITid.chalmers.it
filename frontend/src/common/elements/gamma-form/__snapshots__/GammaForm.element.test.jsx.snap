// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GammaForm/> Shallow render of <GammaForm/> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaForm
    initialValues={
        Object {
            "icecream": "chocolate",
            "mySwitch": false,
            "number": "5000",
            "password": "fdas",
            "user": "asdf",
            "userAgreement": true,
          }
    }
    onSubmit={[Function]}
    render={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "enableReinitialize": false,
      "initialValues": Object {
        "icecream": "chocolate",
        "mySwitch": false,
        "number": "5000",
        "password": "fdas",
        "user": "asdf",
        "userAgreement": true,
      },
      "isInitialValid": false,
      "onSubmit": [Function],
      "render": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
      "validationSchema": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "enableReinitialize": false,
        "initialValues": Object {
          "icecream": "chocolate",
          "mySwitch": false,
          "number": "5000",
          "password": "fdas",
          "user": "asdf",
          "userAgreement": true,
        },
        "isInitialValid": false,
        "onSubmit": [Function],
        "render": [Function],
        "validateOnBlur": true,
        "validateOnChange": true,
        "validationSchema": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
